include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_VIDEO_SUB_ALLOCATED_DESCRIPTOR_SET_H_
+#define _NBL_VIDEO_SUB_ALLOCATED_DESCRIPTOR_SET_H
+
+#include "nbl/video/alloc/IBufferAllocator.h"
+
+#include <type_traits>
+
+namespace nbl::video
+{
+
+// address allocator gives offsets
+// reserved allocator allocates memory to keep the address allocator state inside
+template<class AddrAllocator, class ReservAllocator=core::allocator<uint8_t>>
+class SubAllocatedDescriptorSet : public core::IReferenceCounted 
+{
+  public:
+        using AddressAllocator = AddrAllocator;
+        using ReservedAllocator = ReservAllocator;
+        using size_type = typename AddressAllocator::size_type;
+        using value_type = typename AddressAllocator::size_type;
+        static constexpr value_type invalid_value = AddressAllocator::invalid_address;
+
+        // constructors
+        template<typename... Args>
+        inline SubAllocatedDescriptorSet(const std::span<const video::IGPUDescriptorSetLayout::SBinding> bindings, 
```

Comment:
why didn't you take a layout instead of a span of bindings?

Reply:
Layout doesn't have the list of bindings

Reply:
Last I checked `IGPUDescriptorSetLayout` inherits from `asset::IDescriptorSetLayout` which has the following public methods:
- `uint32_t getTotalBindingCount()`
- `uint32_t getTotalDescriptorCount(const IDescriptor::E_TYPE type)`
- `const CBindingRedirect& getDescriptorRedirect(const IDescriptor::E_TYPE type) const`

What do you need in `CBindingRedirect`  ?  a getter for `CBindingRedirect::m_createFlags` ?

Reply:
Ah ok I think I missed CBindingRedirect